### PR TITLE
#163514587 Create CRUD Operations on Comment

### PR DESCRIPTION
**Pull request by YvesIraguha**
#### What does this PR do?
This PR adds the functionality of updating, fetching, and deleting a comment. 
#### Description of Task to be completed?
* Allow a user to edit the comment
* Allow a user to delete a comment
* Allow a user to read all comments related to a post
#### How should this be manually tested?
* Open Postman client
* Consume the API:
    * `post: /api/v1/post/<postId>/comments` supply the user 's token,and text of the comment.
    * `put: /api/v1/post/<postId>/comments` supply the user 's token, and the updated text.
    * `delete: /api/v1/post/<postId>/comments/<commentId>` supply the user 's token.
    * `get: /api/v1/post/<postId>/comments` supply the user 's token.

#### What are the relevant pivotal tracker stories?
#163514587

### CODE REVIEW

File: controllers/comments.js [hunk 1]
```diff
@@ -16,8 +16,43 @@ export default class Comments {
...
   }
+
+  static async update(req, res) {
+    const id = req.params.commentId;
+    const { text } = req.body;
+    try {
+      const newComment = await Comment.update({ text }, { where: { id } });
+      res.status(201).json({ message: 'Comment updated successfully', newComment });
+    } catch (error) {
+      res.status(400).json(error);
```
Comment (espoirMur) [inline controllers/comments.js]: @YvesIraguha  why the status code 400 for the error??
I think it should be 500, the code 400 is for authentication errors

File: tests/comment.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import request from 'supertest';
```
Comment (espoirMur) [inline tests/comment.test.js:1]: @YvesIraguha, Thanks for taking the initiative of writing tests but while testing you should not only take into consideration the successful case!
- Can you also test what happens when a user tries to comment on a post that doesn't exist?
- Or what happens when the text in the comment is empty?

Please take these case into consideration while testing!
Thanks!

**Review by espoirMur — CHANGES_REQUESTED** [review]
Hey, @YvesIraguha  
Consider updating the status code from 400 to 500 for errors
And if possible consider adding more test cases when testing.....

File: tests/comment.test.js (hunk 2 above)
Comment (YvesIraguha) [reply tests/comment.test.js:1]: Thank you for the input, let me add more test case scenarios.

**Pull request merged**